.github/workflows/docker_build_and_deploy.yml

```diff
@@ -0,0 +1,139 @@
+name: Docker build and deploy
+
+on:
+  push:
+    branches: [ develop, master ]
+  pull_request:
+    branches: [ develop, master ]
+  workflow_dispatch:
+
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+  build-and-deploy:
+    name: Build and deploy image
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+
+    - name: Configure AWS Credentials
+      uses: aws-actions/configure-aws-credentials@v1
+      with:
+        role-to-assume: ${{ secrets.AWS_ROLE_TO_ASSUME }}
+        role-session-name: GithubActionsSession
+        aws-region: us-east-2
+      if: github.event_name != 'pull_request'
+
+    - name: Login to Amazon ECR
```

Comment:
why use amazon ecr instead of a public docker.io account? This is open source and should be free.